internals/cli/masker/writer.go

```diff
@@ -155,49 +166,48 @@ func (mw *MaskedWriter) flushBuffer() {
 //
 // This should be run in a separate goroutine.
 func (mw *MaskedWriter) Run() {
+	go mw.write()
 	masking := false
 	for {
 		select {
-		case <-time.After(mw.timeout):
-			mw.lock.Lock()
-			if len(mw.output) == 0 {
-				for _, matcher := range mw.matchers {
-					matcher.Reset()
-				}
-				mw.flushBuffer()
-			}
-			mw.lock.Unlock()
 		case output := <-mw.output:
 			for _, b := range output {
 				var err error
 				if b.masked {
 					if !masking {
 						_, err = mw.w.Write([]byte(mw.maskString))
 						if err != nil {
-							mw.err = err
+							mw.err <- err
 							return
 						}
 					}
 					masking = true
 				} else {
 					_, err = mw.w.Write([]byte{b.byte})
 					if err != nil {
-						mw.err = err
+						mw.err <- err
 						return
 					}
 					masking = false
 				}
 			}
-			mw.nOut += int64(len(output))
+			mw.wg.Add(-len(output))
```

Comment:
Add a comment to this please (and maybe some whitespace). It is one of the crucial control flows in this entire file. There's only three places where it's used, all in different places: once for adding, once for subtracting, and once for waiting. It's nice if you're able to spot this at a glance.